leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/json/LwM2mNodeJsonEncoder.java

```diff
@@ -198,37 +212,33 @@ public void visit(LwM2mResourceInstance instance) {
                         resourceInstancePath = resourcePath + "/" + entry.getKey();
                     }
 
-                    // Create resource element
-                    JsonArrayEntry jsonResourceElt = new JsonArrayEntry();
-                    jsonResourceElt.setName(resourceInstancePath);
-                    jsonResourceElt.setTime(timestamp);
-
-                    // Convert value using expected type
-                    LwM2mPath lwM2mResourceInstancePath = new LwM2mPath(resourceInstancePath);
-                    Object convertedValue = converter.convertValue(entry.getValue(), resource.getType(), expectedType,
-                            lwM2mResourceInstancePath);
-                    this.setResourceValue(convertedValue, expectedType, jsonResourceElt, lwM2mResourceInstancePath);
-
-                    // Add it to the List
-                    resourcesList.add(jsonResourceElt);
+                    JsonArrayEntry jsonArrayEntry = resourceElement(resourceInstancePath, timestamp,
+                            resource.getType(), expectedType, entry.getValue());
+                    resourcesList.add(jsonArrayEntry);
                 }
             } else {
-                // Create resource element
-                JsonArrayEntry jsonResourceElt = new JsonArrayEntry();
-                jsonResourceElt.setName(resourcePath);
-                jsonResourceElt.setTime(timestamp);
-
-                // Convert value using expected type
-                LwM2mPath lwM2mResourcePath = resourcePath != null ? new LwM2mPath(resourcePath) : null;
-                this.setResourceValue(converter.convertValue(resource.getValue(), resource.getType(), expectedType,
-                        lwM2mResourcePath), expectedType, jsonResourceElt, lwM2mResourcePath);
-
-                // Add it to the List
-                resourcesList.add(jsonResourceElt);
+                JsonArrayEntry jsonArrayEntry = resourceElement(resourcePath, timestamp,
+                        resource.getType(), expectedType, resource.getValue());
+                resourcesList.add(jsonArrayEntry);
             }
             return resourcesList;
         }
 
+        private JsonArrayEntry resourceElement(String resourcePath, Long timestamp, Type type, Type expectedType, Object value) {
```

Comment:
should we rename it in something like :  `createJsonArrayEntry`

Reply:
Done